tests/integration.rs

```diff
@@ -13,58 +13,73 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use risc0_verifier::{verify, VerifyError};
+use risc0_verifier::{verify, DeserializeError, VerifyError};
 use rstest::rstest;
 use serde::Deserialize;
 use std::path::{Path, PathBuf};
 
-fn load_data(path: &Path) -> (Vec<u8>, [u32; 8]) {
+fn load_data(path: &Path) -> ([u32; 8], Vec<u8>, Vec<u8>) {
     #[derive(Deserialize)]
     struct Data {
-        proof_raw_data: String,
-        image_id: [u32; 8],
+        vk: [u32; 8],
+        proof: String,
+        pubs: String,
     }
 
-    let Data {
-        proof_raw_data,
-        image_id,
-    } = serde_json::from_reader(std::fs::File::open(path).unwrap()).unwrap();
+    let Data { vk, proof, pubs } =
+        serde_json::from_reader(std::fs::File::open(path).unwrap()).unwrap();
 
-    let proof_raw_data = <Vec<u8>>::try_from(hex::decode(proof_raw_data).unwrap()).unwrap();
+    let proof = <Vec<u8>>::try_from(hex::decode(proof).unwrap()).unwrap();
+    let pubs = <Vec<u8>>::try_from(hex::decode(pubs).unwrap()).unwrap();
 
-    (proof_raw_data, image_id)
+    (vk, proof, pubs)
 }
 
 #[rstest]
 fn should_verify_valid_proof(#[files("./resources/valid_proof_*.json")] path: PathBuf) {
-    let (proof_raw_data, image_id_data) = load_data(&path);
+    let (vk, proof, pubs) = load_data(&path);
 
-    assert!(verify(&proof_raw_data, image_id_data.into()).is_ok());
+    assert!(verify(vk.into(), &proof, &pubs).is_ok());
 }
 
 #[test]
 fn should_not_verify_invalid_proof() {
```

Comment:
Also here: `should_not_deserialize_invalid_proof()` is a better name.

Reply:
Ok renamed accordingly.